Zotero/Scenes/Detail/DetailCoordinator.swift

```diff
@@ -118,67 +118,135 @@ final class DetailCoordinator: Coordinator {
-        let downloadBatchData = ItemsState.DownloadBatchData(batchData: fileDownloader.batchData)
-        let remoteDownloadBatchData = ItemsState.DownloadBatchData(batchData: remoteFileDownloader.batchData)
-        let identifierLookupBatchData = ItemsState.IdentifierLookupBatchData(batchData: identifierLookupController.batchData)
-        let state = ItemsState(
-            collection: collection,
-            libraryId: libraryId,
-            sortType: .default,
-            searchTerm: searchTerm,
-            filters: [],
-            downloadBatchData: downloadBatchData,
-            remoteDownloadBatchData: remoteDownloadBatchData,
-            identifierLookupBatchData: identifierLookupBatchData,
-            error: nil
-        )
-        let handler = ItemsActionHandler(
-            dbStorage: dbStorage,
-            fileStorage: self.controllers.fileStorage,
-            schemaController: self.controllers.schemaController,
-            urlDetector: self.controllers.urlDetector,
-            fileDownloader: fileDownloader,
-            citationController: citationController,
-            fileCleanupController: fileCleanupController,
-            syncScheduler: syncScheduler,
-            htmlAttributedStringConverter: htmlAttributedStringConverter
-        )
-        let controller = ItemsViewController(viewModel: ViewModel(initialState: state, handler: handler), controllers: self.controllers, coordinatorDelegate: self)
-        controller.tagFilterDelegate = itemsTagFilterDelegate
-        itemsTagFilterDelegate?.delegate = controller
-        return controller
+        func createTrashViewController(
+            libraryId: LibraryIdentifier,
+            dbStorage: DbStorage,
+            fileDownloader: AttachmentDownloader,
+            schemaController: SchemaController,
+            fileStorage: FileStorage,
+            urlDetector: UrlDetector,
+            itemsTagFilterDelegate: ItemsTagFilterDelegate?,
+            htmlAttributedStringConverter: HtmlAttributedStringConverter
+        ) -> TrashViewController {
+            itemsTagFilterDelegate?.clearSelection()
+
+            let searchTerm = searchItemKeys?.joined(separator: " ")
+            let sortType = Defaults.shared.itemsSortType
+            let downloadBatchData = ItemsState.DownloadBatchData(batchData: fileDownloader.batchData)
+            let state = TrashState(libraryId: libraryId, sortType: sortType, searchTerm: searchTerm, filters: [], downloadBatchData: downloadBatchData)
+            let handler = TrashActionHandler(
+                dbStorage: dbStorage,
+                schemaController: schemaController,
+                fileStorage: fileStorage,
+                fileDownloader: userControllers.fileDownloader,
+                urlDetector: urlDetector,
+                htmlAttributedStringConverter: htmlAttributedStringConverter,
+                fileCleanupController: userControllers.fileCleanupController
```

Comment:
Ok I use `controllers` and `userControllers` directly instead of listing everything and I pass instance variables as function params now. `userControllers` is a local variable in this case though.